amkotlinutil/src/main/java/dk/adaptmobile/amkotlinutil/navigation/BaseTab.kt

```diff
@@ -0,0 +1,10 @@
+package dk.adaptmobile.amkotlinutil.navigation
+
+import android.os.Bundle
+import androidx.annotation.IdRes
+
+interface BaseTabObject {
+    fun getInitialView() : BaseView<*, *>
+}
+
+abstract class BaseTab(@IdRes val id: Int, var state: Bundle? = null) : BaseTabObject
```

Comment:
Maybe we can remove BaseTapObject and instead pass in a function to the BaseTab constructor, which returns the view to initialize:
`abstract class BaseTab(@IdRes val id: Int, var initalView: () -> BaseView<*,*>, var state: Bundle? = null)`

Then the usage would change like this,which is cleaner imo. What do you think @martinloesethjensen 
 @GaborHartyandi 

**from**
```
sealed class Tab(id: Int, state: Bundle? = null) : BaseTab(id, state) {
    object Home : Tab(R.id.tab_home) {
        override fun getInitialView(): BaseView<*, *> = HomeView()
    }

    object Search : Tab(R.id.tab_search) {
        override fun getInitialView(): BaseView<*, *> = SearchView()
    }

    object Membercard : Tab(R.id.tab_membercard) {
        override fun getInitialView(): BaseView<*, *> {
            return if (PrefsManager.isUserLoggedIn) {
                MembercardView()
            } else {
                LoginView(LoginType.MemberCardTab)
            }
        }
    }

    object About : Tab(R.id.tab_about) {
        override fun getInitialView(): BaseView<*, *> = AboutView()
    }
}
```

**to:**
```
sealed class Tab(id: Int, state: Bundle? = null, initalView: () -> BaseView<*, *>) : BaseTab(id, initalView, state) {
    object Home : Tab(R.id.tab_home, {HomeView()})

    object Search : Tab(R.id.tab_search, {HomeView()})

    object Membercard : Tab(R.id.tab_membercard, {
        return if (PrefsManager.isUserLoggedIn) {
            MembercardView()
        } else {
            LoginView(LoginType.MemberCardTab)
        }
    })

    object About : Tab(R.id.tab_about, {HomeView()})
}
```